backend/x11/output.c

```diff
@@ -47,8 +47,16 @@ static bool output_set_custom_mode(struct wlr_output *wlr_output,
 	output_set_refresh(&output->wlr_output, refresh);
 
 	const uint32_t values[] = { width, height };
-	xcb_configure_window(x11->xcb_conn, output->win,
+	xcb_void_cookie_t cookie = xcb_configure_window_checked(x11->xcb_conn, output->win,
 		XCB_CONFIG_WINDOW_WIDTH | XCB_CONFIG_WINDOW_HEIGHT, values);
+
+	xcb_generic_error_t *error;
+	if ((error = xcb_request_check(x11->xcb_conn, cookie))) {
+		wlr_log(WLR_ERROR, "Could not set window size to %dx%d\n", width, height);
+		free(error);
+		return false;
```

Comment:
Should this also print information about what the error was?
Also, what kind of errors do you expect? According to https://www.x.org/releases/X11R7.5/doc/x11proto/proto.html (search for `ConfigureWindow`), the possible errors are Match, Value, and Window:

- BadMatch cannot happen here (happen when setting the border width of an InputOnly window or changing the stacking order in an incorrect way).
- BadValue can easily be checked against before sending the broken request (happens when setting width or height to zero).
- BadWindow... well, you have worse problems if you got your own window ID wrong.

So basically, this change introduces an unnecessary round-trip to the X11 server, I think.